exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/internal/OtlpConfigUtil.java

```diff
@@ -237,9 +239,27 @@ public static void setMemoryModeOnOtlpMetricExporterBuilder(
                 "setMemoryMode", MemoryMode.class);
         method.setAccessible(true);
         method.invoke(builder, memoryMode);
+      } else if (builder instanceof OtlpGrpcSpanExporterBuilder) {
+        // Calling getDeclaredMethod causes all private methods to be read, which causes a
+        // ClassNotFoundException when running with the OkHttHttpProvider as the private
+        // setManagedChanel(io.grpc.ManagedChannel) is reached and io.grpc.ManagedChannel is not on
+        // the classpath. io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanUtil provides a layer
+        // of indirection which avoids scanning the OtlpGrpcSpanExporterBuilder private methods.
+        Class<?> otlpGrpcMetricUtil =
+            Class.forName("io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanUtil");
+        Method method =
+            otlpGrpcMetricUtil.getDeclaredMethod(
+                "setMemoryMode", OtlpGrpcSpanExporterBuilder.class, MemoryMode.class);
+        method.setAccessible(true);
+        method.invoke(null, builder, memoryMode);
+      } else if (builder instanceof OtlpHttpSpanExporterBuilder) {
+        Method method =
+            OtlpHttpSpanExporterBuilder.class.getDeclaredMethod("setMemoryMode", MemoryMode.class);
+        method.setAccessible(true);
+        method.invoke(builder, memoryMode);
       } else {
```

Comment:
This ugliness will go away once we promote memory mode to the stable API. I think we just need a release or so to be sure we think this concept is going to work.